subsys/bluetooth/host/buf.c

```diff
@@ -103,6 +108,28 @@ struct net_buf *bt_buf_get_evt(uint8_t evt, bool discardable,
 		buf = net_buf_alloc(&sync_evt_pool, timeout);
 		break;
 #endif /* CONFIG_BT_CONN || CONFIG_BT_ISO */
+	case BT_HCI_EVT_CMD_COMPLETE:
+		__fallthrough;
+	case BT_HCI_EVT_CMD_STATUS:
+		/**
+		 * Use K_NO_WAIT here, to avoiding dead-lock.
+		 * Normally there is always free event buffer corresponding to command.
+		 * When there is no corresponding buf, it usually means that some error
+		 * has occurred.
+		 *
+		 * For example: the Controller spontaneously sends an event.
+		 */
```

Comment:
Is that valid behavior of the controller?

Reply:
there is the command complete for `NOP` opcode. This is legal to send on boot, and we have a quirk in the host to wait for it, as it is somewhat useful over H4.

Reply:
Also: https://github.com/zephyrproject-rtos/zephyr/issues/64158

Reply:
Currently normal event pool will also use by extended advertising event report, this is a huge activity, Also other events such as periodic advertising report. So for command complete and command status also use same normal pool not a good way.